elyra-ai/elyra
# Fix computation of absolute path for url-based KFP components

### What changes were proposed in this pull request?
This fixes a bug where the absolute path was being appended to KFP runtime-specific components for both filepath and url-based components. 

### How was this pull request tested?

- Add a new url-based KFP component to the KFP component catalog file. 
- Add the component to a pipeline and export or submit.
- Export/submission succeeds.

Two new tests added (one with a filename-based component source and one with a url-based source) to confirm that KFP custom components are being successfully converted into their corresponding `TaskSpec` objects.

 
Developer's Certificate of Origin 1.1

       By making a contribution to this project, I certify that:

       (a) The contribution was created in whole or in part by me and I
           have the right to submit it under the Apache License 2.0; or

       (b) The contribution is based upon previous work that, to the best
           of my knowledge, is covered under an appropriate open source
           license and I have the right under that license to submit that
           work with modifications, whether created in whole or in part
           by me, under the same open source license (unless I am
           permitted to submit under a different license), as indicated
           in the file; or

       (c) The contribution was provided directly to me by some other
           person who certified (a), (b) or (c) and I have not modified
           it.

       (d) I understand and agree that this project and the contribution
           are public and that a record of the contribution (including all
           personal information I submit with it, including my sign-off) is
           maintained indefinitely and may be redistributed consistent with
           this project or the open source license(s) involved.
